lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -877,4 +892,79 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     */
+    private function getOrderByForOver(string $query) : string
+    {
+        //determine if 'ORDER BY' is part of the query
+        $orderByPosition = strripos($query, 'order by');
+
+        // early return if ORDER BY not found in query string
+        if ($orderByPosition === false) {
+            return '';
+        }
+
+        // build dictionary if available
+
+        $selectedColumns = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+
+        // filter out 'AS'
+        $filteredSelectedColumns = array_filter(
+            // split selected columns
+            $selectedColumns,
+            function ($element) {
+                // don't return 'AS' and empty elements
+                return strtoupper($element) !== 'AS' && trim($element !== '') && $element !== false;
+            }
+        );
+
+        // re-sequence values
+        $queryArray = array_values($filteredSelectedColumns);
+
+        $orderByArray = explode(',', substr($query, $orderByPosition + strlen('ORDER BY')));
+
+        foreach ($orderByArray as $orderIndex => $orderValue) {
+            $orderByArray[$orderIndex] = implode(
+                ' ',
+                array_filter($this->prepareSplitOrder($orderValue, $queryArray))
+            );
+        }
+
+        $orderByArray[0] = 'ORDER BY ' . $orderByArray[0];
+
+        return implode(',', $orderByArray);
+    }
+
+    /**
+     * Prepare SplitOrder array
+     *
+     * @param string   $orderValue The original ORDER BY string
+     * @param string[] $queryArray Columns from the ORDER BY
+     *
+     * @return string[] New ORDER BY columns
+     */
+    private function prepareSplitOrder($orderValue, $queryArray)
+    {
+        $splitOrder = array_filter(explode(' ', $orderValue));
```

Comment:
Remember you told me that stripping out these thing in an earlier request would be nonsense?

Reply:
I may have said that I'd prefer parsing the `ORDER BY` part of the query as it is using a proper delimiter expression instead of normalizing whitespaces in the entire query (in https://github.com/doctrine/dbal/pull/2879#discussion_r143098611). Do you mean that comment?